cot/src/auth/db.rs

```diff
@@ -115,6 +118,55 @@ impl DatabaseUser {
         Ok(user)
     }
 
+    /// Get a user by their integer ID. Returns [`None`] if the user does not
+    /// exist.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if there was an error querying the database.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the user ID provided is not an integer.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cot::auth::db::DatabaseUser;
+    /// use cot::auth::{Password, UserId};
+    /// use cot::request::{Request, RequestExt};
+    /// use cot::response::{Response, ResponseExt};
+    /// use cot::{Body, StatusCode};
+    ///
+    /// async fn view(request: &Request) -> cot::Result<Response> {
+    ///     let user = DatabaseUser::create_user(
+    ///         request.db(),
+    ///         "testuser".to_string(),
+    ///         &Password::new("password123"),
+    ///     )
+    ///     .await?;
+    ///
+    ///     let user_from_db = DatabaseUser::get_by_id(request.db(), UserId::Int(user.id())).await?;
+    ///
+    ///     Ok(Response::new_html(
+    ///         StatusCode::OK,
+    ///         Body::fixed("User created!"),
+    ///     ))
+    /// }
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> cot::Result<()> {
+    /// #     use cot::test::{TestDatabase, TestRequestBuilder};
+    /// #     let mut test_database = TestDatabase::new_sqlite().await?;
+    /// #     test_database.with_auth().run_migrations().await;
+    /// #     let request = TestRequestBuilder::get("/")
+    /// #         .with_db_auth(test_database.database())
+    /// #         .build();
+    /// #     view(&request).await?;
+    /// #     test_database.cleanup().await?;
+    /// #     Ok(())
+    /// # }
+    /// ```
     pub async fn get_by_id<DB: DatabaseBackend>(db: &DB, id: UserId) -> Result<Option<Self>> {
```

Comment:
On a side note, shouldn't we support getting by String ID?

Reply:
I don't think so, the numeric ID is the primary key in the database and **it** is the unique identifier that's returned by `id()`. I think though that this method should be changed to get `i64` instead of `UserId`. I'll do it in a separate PR.